frontend/src/features/public-form/utils/inputValidation.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  CheckboxFieldValues,
+  MultiAnswerValue,
+  RadioFieldValues,
+  SingleAnswerValue,
+  TableFieldValues,
+  VerifiableFieldValues,
+} from '~templates/Field/types'
+
+/**
+ * Regex to validate yyyy-mm-dd strings.
+ * @note this does not validate that the date is valid, just that the format is correct.
+ */
+const ISO_8601_DATE_ONLY_REGEX =
+  /^\d{4}-([0][1-9]|1[0-2])-([0-2][1-9]|[1-3]0|3[01])$/
+
+export const validateMultiAnswerInput = (
+  input: unknown,
+): input is MultiAnswerValue => {
+  return Array.isArray(input) && input.every((i) => typeof i === 'string')
+}
+
+export const validateTableInput = (
+  input: unknown,
+): input is TableFieldValues => {
+  return (
+    Array.isArray(input) &&
+    input.every((rowData) => {
+      if (typeof rowData !== 'object') return false
+      return Object.keys(rowData).every(
+        (colId) => typeof rowData[colId] === 'string',
+      )
```

Comment:
this will return `true` even if `rowData` is an array right?

Reply:
yea, will add a guard

Reply:
inputValidation utils deleted.